Bugfix setup channels and 2d1d links

shartgring opened the pull request:
## Issue addressed
Fixes #131 and #132 

## Explanation
Some typos and incorrect variable types resulted in errors being raised. This PR solves this by removing the typos and adapting some of the code copied from hydrolib to be able to work with correct variable types. This is not an improvement of the functionality, but merely bigfixing

## Checklist
- [X] Updated tests or added new tests
- [X] Branch is up to date with `main`
- [X] Tests & pre-commit hooks pass
- [x] Updated documentation if needed <-- default for spacing has been set to np.inf, include this in documentation?
- [x] Updated changelog.rst if needed <-- should I add specifics on the bugfixes, or can we call it 'bufixing' in general? Or maybe refer to the methods where the bugs have been fixed?

hboisgon reviewed (approved):
Looks good! For the changelog you can mention the bugfix for direction 2d1d (#132 ). Was the spacing value in any of the hydromt build/update configuration examples? If so it's good to mention. Can you check that this works also with the PR #139 ? You may have conflicts when merging on unifricttype

shartgring commented:
@hboisgon I addressed your points! It will not affect the examples as setup_channels is not included in the examples (thus they are not affected by the changes). I will make an issue that it would be good to include it, as such an example can also be used for tests
